timdex_dataset_api/dataset.py

```diff
@@ -469,3 +389,82 @@ def log_write_statistics(self, start_time: float) -> None:
             f"total rows: {total_rows}, "
             f"total size: {total_size}"
         )
+
+    def read_batches_iter(
```

Comment:
Commenting here but applies to all the `read_*` methods: I appreciate how all the nitty-gritty is localized to the `_get_filtered_dataset` method, clearly distinguishing the actually minimal differences between the `read_*` methods. Nice-nice!

Reply:
Agreed!  It's coming together.